junit-platform-console/src/main/java/org/junit/platform/console/options/AvailableOptions.java

```diff
@@ -10,273 +10,335 @@
+
+	@Option(names = { "-include-package" }, arity = "1", hidden = true)
+	private List<String> includePackages2 = new ArrayList<>();
+
+	@Option(names = {
+			"--exclude-package" }, paramLabel = "PKG", arity = "1", description = "Provide a package to be excluded from the test run. This option can be repeated.")
+	private List<String> excludePackages = new ArrayList<>();
+
+	@Option(names = { "-exclude-package" }, arity = "1", hidden = true)
+	private List<String> excludePackages2 = new ArrayList<>();
+
+	@Option(names = { "-t",
+			"--include-tag" }, paramLabel = "TAG", arity = "1", description = "Provide a tag or tag expression to include only tests whose tags match. "
+					+ //
+					"When this option is repeated, all patterns will be combined using OR semantics.")
+	private List<String> includedTags = new ArrayList<>();
+
+	@Option(names = { "--t", "-include-tag" }, arity = "1", hidden = true)
+	private List<String> includedTags2 = new ArrayList<>();
+
+	@Option(names = { "-T",
+			"--exclude-tag" }, paramLabel = "TAG", arity = "1", description = "Provide a tag or tag expression to exclude those tests whose tags match. "
+					+ //
+					"When this option is repeated, all patterns will be combined using OR semantics.")
+	private List<String> excludedTags = new ArrayList<>();
+
+	@Option(names = { "--T", "-exclude-tag" }, arity = "1", hidden = true)
+	private List<String> excludedTags2 = new ArrayList<>();
+
+	@Option(names = { "-e",
+			"--include-engine" }, paramLabel = "ID", arity = "1", description = "Provide the ID of an engine to be included in the test run. This option can be repeated.")
+	private List<String> includedEngines = new ArrayList<>();
+
+	@Option(names = { "--e", "-include-engine" }, arity = "1", hidden = true)
+	private List<String> includedEngines2 = new ArrayList<>();
+
+	@Option(names = { "-E",
+			"--exclude-engine" }, paramLabel = "ID", arity = "1", description = "Provide the ID of an engine to be excluded from the test run. This option can be repeated.")
+	private List<String> excludedEngines = new ArrayList<>();
+
+	@Option(names = { "--E", "-exclude-engine" }, arity = "1", hidden = true)
+	private List<String> excludedEngines2 = new ArrayList<>();
+
+	// --- Configuration Parameters ----------------------------------------
+
+	// Implementation note: the @Option annotation is on a setter method to allow validation.
+	private Map<String, String> configurationParameters = new LinkedHashMap<>();
+
+	@Spec
+	private CommandSpec spec;
+
+	AvailableOptions() {
+	}
+
+	// CHECKSTYLE:OFF
+	// Checkstyle fails on @throws ParameterException (regardless of fully qualified or not)
+	/**
+	 * Adds the specified key-value pair (or pairs) to the configuration parameters.
+	 * A {@code ParameterException} is thrown if the same key is specified multiple times
+	 * on the command line.
+	 *
+	 * @param map the key-value pairs to add
+	 * @throws picocli.CommandLine.ParameterException if the map already contains this key
```

Comment:
@remkop Is there a reason we need to throw this exact exception type? We usually only throw subclasses of `JUnitException` from our own code.

Reply:
Hi @marcphilipp , thanks for reviewing! 
Yes, the reason is to create a cleaner stack trace.

An exception thrown from the annotated setter method is caught by picocli and rethrown wrapped in a PicocliException, unless the exception is a PicocliException or subclass. This results in multiple nested exceptions that make it difficult for the end user to see the problem.

Please see https://github.com/remkop/picocli/issues/431 for an illustration.

Note that any exceptions thrown by picocli during parsing are caught and rethrown wrapped in a `JUnitException` (see `PicocliCommandLineOptionsParser` line 39). So the top-level exception is always a `JUnitException`.

Reply:
Makes sense, thanks for the explanation! I fixed the Checkstyle issue and pushed an additional commit to your branch.

Reply:
Nice, thank you!